source/fab/dumper.py

```diff
@@ -0,0 +1,76 @@
+##############################################################################
+# (c) Crown copyright Met Office. All rights reserved.
+# For further details please refer to the file COPYRIGHT
+# which you should have received as part of this distribution
+##############################################################################
+"""
+Core of the database dump application.
+"""
+import logging
+from pathlib import Path
+import sys
+
+from fab.database import FileInfoDatabase, SqliteStateDatabase
+from fab.tasks.fortran import FortranWorkingState
+
+
+def entry() -> None:
+    """
+    Entry point for the Fab state database dump tool.
+    """
+    import argparse
+    import fab
+
+    logger = logging.getLogger('fab-dumper')
```

Comment:
There's a bit of mixing of naming, the entry point is referred to as `fab-dumper` in `setup.py` and in the source filename of this new file, but the actual command is called `fab-dump`.  I wonder if perhaps from the user's point of view it makes more sense for the logging identifier to match the name of the command they typed.  (On the other hand, it's not as if it's ambiguous; there is clearly only one command it could be talking about)

Reply:
You've identified a problem but it's the opposite to the one you think it is.

loggers are named in a dot separated hierarchy and messages are passed up the tree until something processes them. When the logger is used it is fetched using `logging.getLogger(__name__)`, i.e. using the package name.

So in order for any logging to ever happen the logger which is set up in the entry point must be on that hierarchy. In other words, it must be named `fab`.

Reply:
Ah okay!  That makes sense